app/helpers/remote_execution_helper.rb

```diff
@@ -60,6 +60,13 @@ def job_invocation_task_buttons(task)
     job_invocation = task.task_groups.find { |group| group.class == JobInvocationTaskGroup }.job_invocation
     task_authorizer = Authorizer.new(User.current, :collection => [task])
     buttons = []
+    template, template_parameters = jobs_template_with_input(job_invocation)
+    if template && authorized_for(controller: :report_templates, action: :generate)
+      buttons << link_to(_('Create Report'), generate_report_template_path(template, template_parameters),
```

Comment:
I'd much rather see it splited into two method calls.
```suggestion
    if (template = job_report_template ) && authorized_for(controller: :report_templates, action: :generate)
      buttons << link_to(_('Create Report'), generate_report_template_path(template, job_report_template_parameters(job_invocation, template)),
```